pkg/instancegroups/instancegroups.go

```diff
@@ -108,11 +108,6 @@ func (c *RollingUpdateCluster) rollingUpdateInstanceGroup(ctx context.Context, c
 	}
 	maxConcurrency := maxSurge + settings.MaxUnavailable.IntValue()
 
-	if maxConcurrency == 0 {
-		klog.Infof("Rolling updates for InstanceGroup %s are disabled", group.InstanceGroup.Name)
-		return nil
```

Comment:
I think apimachinery rules would allow us to fail the rolling update with 0/0 - as long as we don't make valid objects suddenly invalid.

Because we've shipped 0/0 as meaning "no rolling updates", I do think it's more user-friendly here also to treat 0/0 as not enabled, but log a warning / deprecate it.